### ISSUE TITLE
Infinite Loop: Autocorrect fails for Layout/EmptyLinesAroundClassBody and Layout/EmptyLinesAroundModuleBody

### ISSUE THREAD

**Issue by tmertens** [rubocop-hq/rubocop#6732]
When a class or module wrapped in another module ends with multiple `end` keywords collapsed onto the same line (e.g. `end; end`), then the `Layout/EmptyLinesAroundClassBody` and/or `Layout/EmptyLinesAroundModuleBody` will cause rubocop autocorrect to fail in an infinite loop if `EnforcedStyle` is set to `empty_lines_except_namespace` or `ending_only`

--------

## Expected behavior

Rubocop should not loop infinitely.

## Actual behavior

Rubocop encounters an infinite loop:

```
bash-4.4# bundle exec rubocop -a -c .rubocop.yml lib/test.rb 
Inspecting 1 file
C

Offenses:

lib/test.rb:10:1: C: [Corrected] Layout/EmptyLinesAroundModuleBody: Empty line missing at module body end.
      end; end
^
lib/test.rb:10:1: C: [Corrected] Layout/EmptyLinesAroundModuleBody: Extra empty line detected at module body end.

0 files inspected, 2 offenses detected, 2 offenses corrected
Infinite loop detected in /lib/test.rb.
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:270:in `check_for_infinite_loop'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:253:in `block in iterate_until_no_changes'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:252:in `loop'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:252:in `iterate_until_no_changes'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:223:in `do_inspection_loop'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:126:in `block in file_offenses'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:144:in `file_offense_cache'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:124:in `file_offenses'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:112:in `process_file'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:89:in `block in each_inspected_file'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:86:in `each'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:86:in `reduce'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:86:in `each_inspected_file'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:76:in `inspect_files'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/runner.rb:48:in `run'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/cli.rb:174:in `execute_runner'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/cli.rb:75:in `execute_runners'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/lib/rubocop/cli.rb:47:in `run'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/exe/rubocop:13:in `block in <top (required)>'
/usr/lib/ruby/2.5.0/benchmark.rb:308:in `realtime'
/usr/lib/ruby/gems/2.5.0/gems/rubocop-0.63.1/exe/rubocop:12:in `<top (required)>'
/usr/bin/rubocop:23:in `load'
/usr/bin/rubocop:23:in `<top (required)>'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/cli/exec.rb:74:in `load'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/cli/exec.rb:74:in `kernel_load'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/cli/exec.rb:28:in `run'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/cli.rb:463:in `exec'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/cli.rb:27:in `dispatch'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/cli.rb:18:in `start'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/exe/bundle:30:in `block in <top (required)>'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/lib/bundler/friendly_errors.rb:124:in `with_friendly_errors'
/usr/lib/ruby/gems/2.5.0/gems/bundler-1.17.3/exe/bundle:22:in `<top (required)>'
/usr/bin/bundle:23:in `load'
/usr/bin/bundle:23:in `<main>'
```

## Steps to reproduce the problem

Using the following `test.rb` file:

```ruby
# frozen_string_literal: true

module Foo
  module Bar
    module EmptyLinesAroundClassBody

      def foo
        'foo'
      end

    end; end
end

```

And the .rubocop.yml file:
```yaml
AllCops:
  TargetRubyVersion: 2.5
Layout/EmptyLinesAroundClassBody:
  EnforcedStyle: empty_lines_except_namespace
Layout/EmptyLinesAroundModuleBody:
  EnforcedStyle: empty_lines_except_namespace

```

1. Install rubocop version 0.63.1.
2. Run `bundle exec rubocop -a -c .rubocop.yml test.rb 
3. Observe the failure

Note that the infinite loop does not occur when running rubocop without autocorrect.

## RuboCop version

This was tested with rubocop version 0.63.1.

```
$ bundle exec rubocop -V
0.63.1 (using Parser 2.6.0.0, running on ruby 2.5.2 x86_64-linux-musl)
```

Infinite loops were also encountered with versions going back to ~0.55 (I didn't test even older versions) but I cannot confirm if it was this specific issue or a different problem as I hadn't identified the offending linter rule yet.
